sw/device/tests/sim_dv/rv_dm_ndm_reset_req.c

```diff
@@ -107,33 +106,24 @@ static test_register_t kReg[] = {
             .base = TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR,
             .offset = OTP_CTRL_DIRECT_ACCESS_WDATA_0_REG_OFFSET,
             .write_val = 0x06092022,
-            .exp_read_val = 0x06092022,
+            .exp_read_val = OTP_CTRL_DIRECT_ACCESS_WDATA_0_REG_RESVAL,
```

Comment:
@sriyerg i feel like this would be better done on the SV side..as in ...we randomly pick some registers to read / write, and make sure we don't pick ones in the `rv_dm` domain.  What do you think? This is pretty hardcoded right now..

Reply:
Agreed - these hardcoded values add a maintenance overhead. We can move it to the SV side. But, are you suggesting having the SV provide the address/write value/check mask to the SW or are you suggesting using a stub cpu mode instead?

Reply:
hmmm.... i'll admit i was thikning of just doing csr backdoor reads hehe.  Is that bad you think?

Reply:
given your explanation about backdoors, i guess maybe this can made into a stub test only long term.
There isn't that much C code.

Reply:
Be careful to honor things like "fetch_en" which should also prevent the STUB cpu model from issuing bus transactions.

Reply:
i don't think that's actually what we do right now. In fact we have specific tests that uses the STUB to get around that to check for specific things. 

@weicaiyang / @msfschaffner know better.

Reply:
IMO, frontdoor is better than backdoor if possible, in case something in the ibex prevents from accessing the CSRs. If we want to improve this, perhaps we can add backdoor read to check all CSRs in the DV sequence.

Reply:
I agree - better to do accesses via frontdoor, and in a non-stub-CPU mode environment. Our best bet would be to randomize {addr, wr_data, exp_rd_data} for each peripheral under-test from the SV side leveraging our SV RAL model's strengths and pass them on to the C side via our regular backdoor write mechanisms. It will probably not be too hard to do this.

Reply:
sounds good, i'll file an issue for the future.